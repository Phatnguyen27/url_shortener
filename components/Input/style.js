import styled from 'styled-components';

export const Wrapper = styled.input`
    height: ${(props) => props?.theme?.input?.height?.value};
    font-size: ${(props) => props?.theme?.input?.fontSize?.value};
    width: ${(props) => props?.theme?.input?.width?.value};
`;
