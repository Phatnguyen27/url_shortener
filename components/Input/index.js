import React from 'react';
import * as style from './style';

const Input = (props) => {
    const {value,onTextChangeHandler,name,id} = props;
    return <style.Wrapper name={name} id={id} type="text" onChange={(event) => {onTextChangeHandler(event.target.value)}}></style.Wrapper>
};

export default Input;