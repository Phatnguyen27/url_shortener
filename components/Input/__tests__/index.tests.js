import React from "react";
import renderer from "react-test-renderer";
import Input from "..";

const testProps = {
    value: "test",
    type: "text",
    placeholder: "test",
    onChange: {
        onChange: jest.fn(() => ({}))
    }
}
const component = renderer.create(
    <Input {...testProps}/>
);
const tree = component.toJSON();

test('<Input />', () => {
    expect(tree).toMatchSnapshot();
});