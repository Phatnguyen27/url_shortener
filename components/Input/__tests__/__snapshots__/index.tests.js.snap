// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> 1`] = `
<input
  className="sc-bdfBwQ leIcBc"
  onChange={[Function]}
  type="text"
/>
`;
